dpctl/_memory.pyx

```diff
@@ -30,33 +30,171 @@
 
 import dpctl
 from dpctl._backend cimport *
-from ._sycl_core cimport SyclContext, SyclQueue
+from ._sycl_core cimport SyclContext, SyclQueue, SyclDevice
+from ._sycl_core cimport get_current_queue
 
 from cpython cimport Py_buffer
+from cpython.bytes cimport PyBytes_AS_STRING, PyBytes_FromStringAndSize
 
+import numpy as np
 
-cdef class Memory:
+cdef _throw_sycl_usm_ary_iface():
+    raise ValueError("__sycl_usm_array_interface__ is malformed")
 
-    cdef _cinit(self, Py_ssize_t nbytes, ptr_type, SyclQueue queue):
-        cdef DPPLSyclUSMRef p
 
+cdef void copy_via_host(void *dest_ptr, SyclQueue dest_queue,
```

Comment:
If the memcpy calls are changed to non-blocking, this function should also be made non-blocking?

Reply:
Yes. It could return an event, but copying from host to device depends on the copying from device to host to complete, so we will need to somehow reflect that. 

Perhaps `copy_via_host` belongs in C++, in SYCLinterface to make this easier to implement.